neps/nep-0448.md

```diff
@@ -0,0 +1,92 @@
+
+Ideally a new user should be able to onboard onto NEAR through any applications built on top of NEAR and do not have to
+understand that the application is running on top of blockchain. The ideal flow is as follows: a user hear about an interesting
+application from their friends or some forum and they decide to give it a try. The user opens the application in their
+browser and directly starts using it without worrying about registration. Under the hood, a keypair is generated for the
+user and the application creates an account for the user and pays for transaction fees through meta transactions. Later on,
+the user may find other applications that they are also interested in and give them a try as well. At some point, the user
+graduates from the onboarding experience by acquiring NEAR tokens either through earning or because they like some experience
+so much that they would like to pay for it explicitly.
+
+## Rationale and alternatives
+
+There are a few alternative ideas:
+* Completely disregard storage staking and do not change the account creation cost. This makes the implementation even
+simpler. However, there may be a risk of spamming attack given that the cost of creating an account is around 0.2Tgas.
+In addition, with the current design, it is easy to further reduce the cost. Going the other way is more difficult.
+* Do not change how storage staking is calculated when converting to gas cost. This means that account creation cost would
+be around 60Tgas, which is both high in gas (meaning that the throughput is limited and more likely for some contract to break)
+and more costly for users (around 0.006N per account creation).
+
+## Specification
+
+There are two main changes to the protocol:
+* Account creation cost needs to be increased. For every account, at creation time, 600 bytes of storage are reserved
+for the account itself + three keys. For function call access keys, the "free" ones cannot use `method_names` in order to
+reserve space. The cost of these bytes is paid through transaction fee. Note that there is already [discussion](https://github.com/near/NEPs/issues/415)
+around the storage cost of NEAR and whether it is reasonable. While this proposal does not attempt to change the entire
+storage staking mechanism, the cost of storage is reduced in 10x when converting to gas. A [discussion](https://gov.near.org/t/storage-staking-price/399)
+from a while ago mentioned this idea, and the concerns there were proven to be not real concerns. No one is deleting
+data from storage in practice and the storage staking mechanism does not really serve its purpose. That conversion means
+we increase the account creation cost to 6Tgas from 0.2Tgas
+* Storage staking check will not be applied if an account has <= 3 keys and does not have a contract deployed. If an
+account accrues more than 3 keys, however, it must pay for the storage of everything including those 3 keys. This makes
+the implementation simpler and less error-prone.
```

Comment:
Some questions that have been answered before but should be answered here as well:

- What happens after a fourth key is added or a contract is deployed? Are all the previous keys counted toward the storage staking?
- What happens if an account has four keys and no contract, and suddenly a key is removed? Is it acceptable for it to have 0N for staking again? 
- Depending on the answer to the previous question, we should take care of the case where a contract modifies the state but is later removed without zeroing the state back.

Reply:
Will update the proposal to be more explicit, but basically it is a binary thing: either an account has free storage for up to 3 keys or it has to pay for everything (including those keys that were "free" earlier).

Reply:
@mfornet I added a more detailed description in the "reference implementation" section. PTAL